Svc/DpCatalog/DpCatalog.cpp

```diff
@@ -204,15 +207,15 @@ namespace Svc {
                     this->log_WARNING_HI_FileHdrDesError(fullFile, desStat);
                 }
 
-                // add entry to catalog. 
+                // add entry to catalog.
                 DpStateEntry entry;
-                entry.dir = dir;
+                entry.dir = static_cast<FwIndexType>(dir);
```

Comment:
**For future review:** why is `entry.dir` `FwIndexType` here and `FwSizeType` in the iterator.